src/test/java/ru/nmedvedev/service/spendmoneyreminder/SpendMoneyReminderServiceTest.java

```diff
@@ -0,0 +1,90 @@
+package ru.nmedvedev.service.spendmoneyreminder;
+
+import io.smallrye.mutiny.Multi;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import ru.nmedvedev.model.HistoryDb;
+import ru.nmedvedev.model.UserDb;
+import ru.nmedvedev.repository.UserRepository;
+import ru.nmedvedev.service.TelegramService;
+import ru.nmedvedev.view.ReplyButtonsProvider;
+
+import java.util.Random;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class SpendMoneyReminderServiceTest {
+
+    @InjectMocks
+    private SpendMoneyReminderService service;
+
+    @Mock
+    private RemindDayProviderService remindDayProviderService;
+    @Mock
+    private SpendMoneyReminderBusinessLogicGateService spendMoneyReminderBusinessLogicGateService;
+    @Mock
+    private UserRepository userRepository;
+    @Mock
+    private TelegramService telegramService;
+    @Mock
+    private ReplyButtonsProvider replyButtonsProvider;
+
+    private static final Long CHAT_ID = new Random().nextLong();
+
+    @Test
+    void doNotFireAnyReminderOnAWrongDay() {
+        when(remindDayProviderService.getDay(any())).thenReturn(ReminderDayEnum.NOT_A_DAY_FOR_A_REMINDER);
+        service.sendReminders();
+        verifyNoInteractions(spendMoneyReminderBusinessLogicGateService);
+        verifyNoInteractions(userRepository);
+        verifyNoInteractions(telegramService);
+        verifyNoInteractions(replyButtonsProvider);
+    }
+
+    @Test
+    void doNotSendIfUserHasNoAmount() {
```

Comment:
Did you mean `NoCard`?
```suggestion
    void doNotSendIfUserHasNoCard() {
```